### PR TITLE
GH-1375: Responsive design for Hub, GH-1407: Minor fix for image display on FF

### PR DESCRIPTION
**Pull request by Eden12345**
JIRA Tickets Resolved:
- Responsive Design for Hub: https://cliqztix.atlassian.net/browse/GH-1375
- Logo Display Bug on Firefox: https://cliqztix.atlassian.net/browse/GH-1407

Added:
- Ghostery Lite link to iTunes App Store in ProductsView

### CODE REVIEW

File: app/hub/Views/SetupViews/SetupAntiSuiteView/SetupAntiSuiteView.jsx [hunk 1]
```diff
@@ -31,28 +31,41 @@ const SetupAntiSuiteView = props => (
...
-						<div className="columns small-12 large-8">
+						<div className="SetupAntiSuite--alignOnMedium columns small-12 large-8">
 							<div className="flex-container align-middle">
-								<div className="SetupAntiSuite__feature_title display-inline">
+								<div className="SetupAntiSuite__featureTitle display-inline">
 									{feature.name}
 								</div>
 								{feature.enabled && (
-									<div className="SetupAntiSuite__feature_enabled display-inline">
+									<div className="SetupAntiSuite__featureEnabled display-inline hide-for-small show-for-large">
```
Comment (IAmThePan) [inline app/hub/Views/SetupViews/SetupAntiSuiteView/SetupAntiSuiteView.jsx]: you don't need hide-for-small

File: app/hub/Views/SideNavigationView/SideNavigationViewContainer.jsx [hunk 2]
```diff
@@ -66,7 +65,12 @@ class SideNavigationViewContainer extends Component {
 			{ id: 'logout', text: t('hub_side_navigation_log_out'), clickHandler: this._handleLogoutClick },
 		] : [
 			{ id: 'create-account', href: '/create-account', text: t('hub_side_navigation_create_account') },
-			{ id: 'log-id', href: '/log-in', text: t('hub_side_navigation_log_in') },
+			{
```
Comment (IAmThePan) [inline app/hub/Views/SideNavigationView/SideNavigationViewContainer.jsx]: make this line consistent with the other items in the array (one line vs multiple)

File: app/hub/Views/SideNavigationView/SideNavigationViewContainer.jsx [hunk 3]
```diff
@@ -47,7 +46,7 @@ class SideNavigationViewContainer extends Component {
 	 */
 	render() {
 		const { user, location } = this.props;
-		const disableRegEx = /^(\/setup)|(\/tutorial)/;
+		const disableRegEx = /^(\/setup(?!\/4))|(\/tutorial(?!\/6))/;
```
Comment (IAmThePan) [inline app/hub/Views/SideNavigationView/SideNavigationViewContainer.jsx]: put a `$` character after 4 and 6 to make sure that it is matching to the last character of the route

File: app/hub/Views/SupporterView/SupporterView.jsx [hunk 4]
```diff
@@ -126,30 +125,51 @@ class SupporterView extends Component {
...
-						<div className="SupporterView__feature columns small-12 medium-5 medium-offset-1">
+						<div className="SupporterView__feature columns large-offset-1">
 							<div className="SupporterView__headingTitle">
 								{t('hub_supporter_feature_support_title')}
 							</div>
 							<div className="SupporterView__headingDescription">
 								{t('hub_supporter_feature_support_description')}
 							</div>
-							{this._renderButton(false)}
+							{this._renderButton('hide-for-small show-for-large')}
```
Comment (IAmThePan) [inline app/hub/Views/SupporterView/SupporterView.jsx]: we don't need both hide-for-small and show-for-large

File: app/images/hub/side-navigation/ghostery-stacked-old.svg [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (IAmThePan) [inline app/images/hub/side-navigation/ghostery-stacked-old.svg]: we don't use this file. please delete

File: app/hub/Views/TutorialView/__tests__/TutorialView.test.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import TutorialView from '../TutorialView';
+
+// Mock Necessary Imports
+jest.mock('../../TutorialViews/TutorialNavigation', () => props => <div>Mock Tutorial Navigation</div>);
+
+describe('app/hub/Views/TutorialView component', () => {
+	describe('Snapshot tests with react-test-renderer', () => {
+		test('tutorial view is rendered correctly', () => {
+			const initialState = {
+				steps: [],
```
Comment (IAmThePan) [inline app/hub/Views/TutorialView/__tests__/TutorialView.test.jsx]: include actual steps.  Look at /app/hub/Views/SetupView/__tests__/SetupView.test.jsx

File: app/hub/Views/AppView/AppView.jsx [hunk 7]
```diff
@@ -21,15 +21,30 @@ import { ToastMessage } from '../../../shared-components';
...
+	constructor(props) {
+		super(props);
+		this.mainContent = React.createRef();
+	}
+
+	componentDidUpdate() {
+		this.mainContent.current.scroll(0, 0);
+	}
+
+	render() {
```
[the remark this answers is not in the document]
Comment (Eden12345) [reply app/hub/Views/AppView/AppView.jsx]: 👍

File: app/hub/Views/AppView/AppView.jsx [hunk 8]
```diff
@@ -21,15 +21,30 @@ import { ToastMessage } from '../../../shared-components';
...
-		</div>
-	</div>
-);
+class AppView extends Component {
+	constructor(props) {
+		super(props);
+		this.mainContent = React.createRef();
+	}
+
+	componentDidUpdate() {
```
[the remark this answers is not in the document]
Comment (Eden12345) [reply app/hub/Views/AppView/AppView.jsx]: 👍

File: app/hub/Views/AppView/__tests__/AppView.test.jsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+					toastMessage: 'Example toast message',
+					toastClass: 'toast-class',
+				},
+				exitToast: () => {},
+			};
+
+			const component = shallow(<AppView {...initialState} />);
+			expect(component.find('.App').length).toBe(1);
+			expect(component.find('.App__mainContent').length).toBe(1);
+		});
```
[the remark this answers is not in the document]
Comment (Eden12345) [reply app/hub/Views/AppView/__tests__/AppView.test.jsx]: Jest doesn't like the scroll function call on the mainContent ref; will circle back to this

File: app/hub/Views/SideNavigationView/SideNavigationViewContainer.jsx (hunk 2 above)
Comment (Eden12345) [reply app/hub/Views/SideNavigationView/SideNavigationViewContainer.jsx]: there's too many key/value pairs for a one liner according to our linter

**Comment by jsignanini** [comment]
@Eden12345 looks great, only 2 things:
1. The modal that shows up on the Hub when you attempt to go through the tutorial after the initial install should also be made responsive.
2. The entire Hub page should have a `<meta name="viewport" ...>` tag to allow it to shrink based on device-width, zoom, etc. See `app/templates/panel_android.html`.

**Comment by Eden12345** [comment]
Added in José's edits and got them approved. Full coverage on JSX tests (not on actions and reducers, per Caleb's instructions). PR should be ready to go.

**Pull request merged**